javascript/javascript1/week4/js1-w4.js

```diff
@@ -0,0 +1,85 @@
+// Suman Ghimire js1 w4
+
+let splitArray = [];
+let theNameArray = [];
+let toDoList = [];
+const calculateArray = [];
+function getReply(command) {
+    let splitText = command.split(` `);
+    if (command.includes('Hello my name is') || command.includes('my name') || command.includes('i am')) {
+        splitArray.push(splitText);
+        let result = "";
+        let theName = "";
+        for (let name of splitArray) {
+            result = name;
+            theName = result[result.length - 1];
+            theNameArray.push(theName);
+        }
+        function userName() {
+            let result = "Nice to meet you " + theName;
+            console.log(result);
+        }
+        userName();
+    } else if (command.includes("What is your name") && theNameArray.length >= 1) {
+        function userName1() {
+            let sentence = "Your name is " + theNameArray[0];
+            console.log(sentence);
+        } userName1();
+    } else if (command == "" || theNameArray.length == 0) {
+        console.log("Your have mention your good name first");
```

Comment:
Ok, but we can't really trust the user's input to be in the exact index that we plan, can we? What if the user writes something like "My name is Harold, good morning". It's better to use template literals so the function will always retrieve the name wherever it is located. ${name}

Reply:
ok. managed to refactor it.